### ISSUE TITLE
[syzkaller] INFO: task hung in lock_sock_nested

### ISSUE THREAD

**Issue by cpaasch** [multipath-tcp/mptcp_net-next#12]
syzkaller triggered on top of 6fe9a949d3fe with Florian's patch
[patch.txt](https://github.com/multipath-tcp/mptcp_net-next/files/4417665/patch.txt)

```
INFO: task syz-executor.4:22557 blocked for more than 143 seconds.
      Not tainted 5.6.0 #64
"echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
syz-executor.4  D    0 22557   2905 0x00000004
Call Trace:
 context_switch kernel/sched/core.c:3380 [inline]
 __schedule+0x23c/0x5f0 kernel/sched/core.c:4080
 schedule+0x4a/0x100 kernel/sched/core.c:4154
 __lock_sock+0x80/0xd0 net/core/sock.c:2424
 lock_sock_nested+0x77/0x80 net/core/sock.c:2949
 lock_sock include/net/sock.h:1574 [inline]
 inet_stream_connect+0x27/0x60 net/ipv4/af_inet.c:718
 mptcp_stream_connect+0xad/0x130 net/mptcp/protocol.c:1658
 __sys_connect_file net/socket.c:1859 [inline]
 __sys_connect+0x140/0x180 net/socket.c:1876
 __do_sys_connect net/socket.c:1887 [inline]
 __se_sys_connect net/socket.c:1884 [inline]
 __x64_sys_connect+0x1e/0x30 net/socket.c:1884
 do_syscall_64+0x91/0x2f0 arch/x86/entry/common.c:294
 entry_SYSCALL_64_after_hwframe+0x44/0xa9
RIP: 0033:0x7f60af9cc469
Code: Bad RIP value.
RSP: 002b:00007f60b00bcdd8 EFLAGS: 00000246 ORIG_RAX: 000000000000002a
RAX: ffffffffffffffda RBX: 000000000066bf00 RCX: 00007f60af9cc469
RDX: 000000000000006e RSI: 0000000020000040 RDI: 0000000000000008
RBP: 00000000ffffffff R08: 0000000000000000 R09: 0000000000000000
R10: 0000000000000000 R11: 0000000000000246 R12: 0000000000000094
R13: 00000000004142ac R14: 00007f60b00bd5c0 R15: 0000000000000003
NMI backtrace for cpu 0
CPU: 0 PID: 484 Comm: khungtaskd Not tainted 5.6.0 #64
Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS rel-1.12.1-0-ga5cab58e9a3f-prebuilt.qemu.org 04/01/2014
Call Trace:
 __dump_stack lib/dump_stack.c:77 [inline]
 dump_stack+0xda/0x116 lib/dump_stack.c:118
 nmi_cpu_backtrace.cold+0x18/0x64 lib/nmi_backtrace.c:101
 nmi_trigger_cpumask_backtrace+0x158/0x191 lib/nmi_backtrace.c:62
 arch_trigger_cpumask_backtrace+0x14/0x20 arch/x86/kernel/apic/hw_nmi.c:38
 trigger_all_cpu_backtrace include/linux/nmi.h:146 [inline]
 check_hung_uninterruptible_tasks kernel/hung_task.c:205 [inline]
 watchdog+0x5f7/0x750 kernel/hung_task.c:289
 kthread+0x153/0x180 kernel/kthread.c:255
 ret_from_fork+0x1f/0x30 arch/x86/entry/entry_64.S:352
Sending NMI from CPU 0 to CPUs 1:
NMI backtrace for cpu 1
CPU: 1 PID: 1589 Comm: systemd-journal Not tainted 5.6.0 #64
Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS rel-1.12.1-0-ga5cab58e9a3f-prebuilt.qemu.org 04/01/2014
RIP: 0010:queued_write_lock_slowpath+0x40/0xa0 kernel/locking/qrwlock.c:77
Code: 0f b1 57 04 75 61 8b 03 85 c0 74 4c f0 81 03 00 01 00 00 b9 ff 00 00 00 be 00 01 00 00 8b 03 3d 00 01 00 00 74 0c f3 90 8b 13 <81> fa 00 01 00 00 75 f4 89 f0 f0 0f b1 0b 3d 00 01 00 00 75 de 48
RSP: 0018:ffffc900001abd78 EFLAGS: 00000006
RAX: 0000000000000300 RBX: ffff888139d60420 RCX: 00000000000000ff
RDX: 0000000000000300 RSI: 0000000000000100 RDI: ffff888139d60420
RBP: ffffc900001abd88 R08: ffff88813b3d9140 R09: 0000000000000000
R10: ffffc90000087e58 R11: 0000000000000001 R12: ffff888139d60424
R13: 0000000000000000 R14: ffff888139d60410 R15: ffff888139d60420
FS:  00007ff97c2a98c0(0000) GS:ffff88813bd00000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 00007ff978dac028 CR3: 000000013901e003 CR4: 0000000000160ee0
Call Trace:
 queued_write_lock include/asm-generic/qrwlock.h:95 [inline]
 __raw_write_lock_irq include/linux/rwlock_api_smp.h:197 [inline]
 _raw_write_lock_irq+0x41/0x50 kernel/locking/spinlock.c:311
 ep_scan_ready_list.constprop.0+0x68/0x210 fs/eventpoll.c:684
 ep_send_events fs/eventpoll.c:1766 [inline]
 ep_poll+0xbf/0x600 fs/eventpoll.c:1903
 do_epoll_wait+0x130/0x150 fs/eventpoll.c:2298
 __do_sys_epoll_wait fs/eventpoll.c:2308 [inline]
 __se_sys_epoll_wait fs/eventpoll.c:2305 [inline]
 __x64_sys_epoll_wait+0x22/0x30 fs/eventpoll.c:2305
 do_syscall_64+0x91/0x2f0 arch/x86/entry/common.c:294
 entry_SYSCALL_64_after_hwframe+0x44/0xa9
RIP: 0033:0x7ff97b573303
Code: 49 89 ca b8 e8 00 00 00 0f 05 48 3d 01 f0 ff ff 73 34 c3 48 83 ec 08 e8 0b c2 00 00 48 89 04 24 49 89 ca b8 e8 00 00 00 0f 05 <48> 8b 3c 24 48 89 c2 e8 51 c2 00 00 48 89 d0 48 83 c4 08 48 3d 01
RSP: 002b:00007ffd617ccd10 EFLAGS: 00000293 ORIG_RAX: 00000000000000e8
RAX: ffffffffffffffda RBX: 0000560192d9f1e0 RCX: 00007ff97b573303
RDX: 0000000000000014 RSI: 00007ffd617ccd20 RDI: 0000000000000008
RBP: 00007ffd617ccf10 R08: 0000000000027c8a R09: 000000007735961f
R10: 00000000ffffffff R11: 0000000000000293 R12: 00007ffd617ccd20
R13: 0000000000000001 R14: ffffffffffffffff R15: 00071afd69ee9dd5
```

syzkaller-repro

```
# {Threaded:false Collide:false Repeat:false RepeatTimes:0 Procs:1 Sandbox: Fault:false FaultCall:-1 FaultNth:0 Leak:false NetInjection:false NetDevices:false NetReset:false Cgroups:false BinfmtMisc:false CloseFDs:false KCSAN:false DevlinkPCI:false UseTmpDir:false HandleSegv:false Repro:false Trace:false}
r0 = socket$inet_mptcp(0x2, 0x1, 0x106)
bind$inet(r0, &(0x7f00000013c0)={0x2, 0x4e20}, 0x10)
listen(r0, 0x0)
r1 = socket$inet_mptcp(0x2, 0x1, 0x106)
connect$inet(r1, &(0x7f0000000040)={0x2, 0x4e20, @loopback}, 0x4d)
r2 = accept(r0, 0x0, 0x0)
connect$unix(r2, 0x0, 0x0)
```

C-repro
```
// autogenerated by syzkaller (https://github.com/google/syzkaller)

#define _GNU_SOURCE

#include <endian.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/syscall.h>
#include <sys/types.h>
#include <unistd.h>

uint64_t r[3] = {0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff};

int main(void)
{
  syscall(__NR_mmap, 0x20000000ul, 0x1000000ul, 3ul, 0x32ul, -1, 0ul);
  intptr_t res = 0;
  res = syscall(__NR_socket, 2ul, 1ul, 0x106);
  if (res != -1)
    r[0] = res;
  *(uint16_t*)0x200013c0 = 2;
  *(uint16_t*)0x200013c2 = htobe16(0x4e20);
  *(uint32_t*)0x200013c4 = htobe32(0);
  syscall(__NR_bind, r[0], 0x200013c0ul, 0x10ul);
  syscall(__NR_listen, r[0], 0);
  res = syscall(__NR_socket, 2ul, 1ul, 0x106);
  if (res != -1)
    r[1] = res;
  *(uint16_t*)0x20000040 = 2;
  *(uint16_t*)0x20000042 = htobe16(0x4e20);
  *(uint32_t*)0x20000044 = htobe32(0x7f000001);
  syscall(__NR_connect, r[1], 0x20000040ul, 0x4dul);
  res = syscall(__NR_accept, r[0], 0ul, 0ul);
  if (res != -1)
    r[2] = res;
  syscall(__NR_connect, r[2], 0ul, 0ul);
  return 0;
}
```

**Comment by pabeni** [comment]
looks like this one is fixed on top the current export branch + "mptcp: fix double-unlock in mptcp_poll" (-net commit e154659ba39a1c2be576aaa0a5bda8088d707950), can you please confirm?

**Comment by cpaasch** [comment]
Yes, no more panic with the reproducer. Closing!

**Comment by cpaasch** [comment]
Actually, no - I didn't wait long enough for the task-hung to be detected.

I still got the crash on commit e154659ba39a ("mptcp: fix double-unlock in mptcp_poll")

**Comment by pabeni** [comment]
I can't reproduce the issue on my local tree (which comprise -net fixes up to 1200832c6e850a17f36631f6492f953a1b39e6b8 plus a pending one). Is the splat backtrace unchanged? how many vCPUs has the VM under test?

**Comment by cpaasch** [comment]
Yes, cant' repro. Closing!
